repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/samlpolicy/SamlFlow20Test.groovy

```diff
@@ -0,0 +1,421 @@
+        deproxy.addEndpoint(properties.targetPort, 'origin service', null, fakeIdentityV2Service.handler)
+        deproxy.addEndpoint(properties.identityPort, 'identity service', null, fakeIdentityV2Service.handler)
+
+        repose.start()
+        reposeLogSearch.awaitByString("Repose ready", 1, 30)
+
+        fakeIdentityV2Service.admin_token = UUID.randomUUID().toString()
+    }
+
+    def setup() {
+        fakeIdentityV2Service.resetCounts()
+        fakeIdentityV2Service.resetHandlers()
+    }
+
+    @Unroll
+    @FailsWith(ConditionNotSatisfiedError)
+    def "a saml:response that is #signatureStatus will still be successfully processed as long as its Assertion is signed"() {
+        when:
+        def mc = deproxy.makeRequest(
+                url: reposeEndpoint + SAML_AUTH_URL,
+                method: HTTP_POST,
+                headers: [(CONTENT_TYPE): CONTENT_TYPE_FORM_URLENCODED],
+                requestBody: asUrlEncodedForm((PARAM_SAML_RESPONSE): encodeBase64(saml)))
+
+        then: "the client gets back a good response"
+        mc.receivedResponse.code as Integer == SC_OK
+
+        and: "the origin service received the request with the correct header values"
+        mc.handlings[0]
+        mc.handlings[0].request.headers.getFirstValue(CONTENT_TYPE) == CONTENT_TYPE_XML
+        mc.handlings[0].request.headers.getFirstValue(IDENTITY_API_VERSION) == "2.0"
+        fakeIdentityV2Service.getGenerateTokenFromSamlResponseCount() == 1
+
+        //and: "the response from the origin service is appropriately updated by the filter"
+        // TODO: what is Flow 2.0 going to do to the response?
+
+        when: "the saml:response received by the origin service is unmarshalled"
+        Response response = samlUtilities.unmarshallResponse(mc.handlings[0].request.body as String)
+
+        then: "there should be two assertions"
+        response.assertions.size() == 2
+
+        and: "the first assertion and the saml:response should indicate that Repose is the Issuer"
+        response.assertions[0].issuer.value == SAML_REPOSE_ISSUER
+        response.issuer.value == SAML_REPOSE_ISSUER
+
+        and: "the second Assertion should still have a valid signature and have the original Issuer"
+        response.assertions[1].issuer.value == SAML_EXTERNAL_ISSUER
+        samlUtilities.validateSignature(response.assertions[1].signature)
+
+        and: "the saml:response should have a valid signature since Repose should have signed it"
+        samlUtilities.validateSignature(response.signature)
+
+        where:
+        signatureStatus    | saml
+        "not signed"       | SAML_ONE_ASSERTION_SIGNED
+        "signed"           | SAML_ASSERTION_AND_MESSAGE_SIGNED
+        "signed (invalid)" | SAML_ASSERTION_AND_MESSAGE_SIGNED.replaceFirst("\n", "").replaceFirst("\n", "")
+    }
+
+    def "a saml:response without an assertion should be rejected"() {
```

Comment:
I have no idea where this would break or if it would at all. Is this a strict requirement that i didn't notice? I mean it's obviously an implied need, but...